### PR TITLE
concatMap for SignalProducers

### PR DESCRIPTION
**Pull request by saniul**
Fixes https://github.com/ReactiveCocoa/ReactiveCocoa/issues/1748

... I think?

### CODE REVIEW

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 1]
```diff
@@ -527,10 +527,16 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+/// Maps each event from `producer` sends to a new producer, then
+/// concatenates the resulting producers together.
+public func concatMap<T, U, E>(transform: T -> SignalProducer<U, E>)(producer: SignalProducer<T, E>) -> SignalProducer<U, E> {
+
```
Comment (kastiglione) [inline ReactiveCocoa/Swift/SignalProducer.swift]: Style: RAC code doesn't use an empty line a the start of methods.

File: ReactiveCocoa/Swift/SignalProducer.swift [hunk 2]
```diff
@@ -527,10 +527,16 @@ public func concat<T, E>(next: SignalProducer<T, E>)(producer: SignalProducer<T,
 	return SignalProducer(values: [producer, next]) |> concat
 }
 
+/// Maps each event from `producer` sends to a new producer, then
```
Comment (kastiglione) [inline ReactiveCocoa/Swift/SignalProducer.swift]: The word "sends" doesn't fit.

**Comment by NachoSoto** [comment]
That looks correct :)

**Comment by jspahrsummers** [comment]
Is this ready for a merge? (We usually comment to indicate that something is ready for another review or merge, since commits don't generate notifications.)

**Comment by saniul** [comment]
Yep, ready for merge!

**Comment by jspahrsummers** [comment]
Thanks for the pull request! :sparkles:

**Pull request merged**